### PR TITLE
Fixes and update

### PR DESCRIPTION
**Pull request by Jiltseb**
Changes:
1. Update to the latest faster-whisper from SYSTRAN.
2. Modified type annotations for transcribe function in batched mode, and added docstring.
3. Handle no vad_segments.
4. Removed unnecessary data classes.
5. Make default_batched_asr_options accessible from the package.

### CODE REVIEW

File: faster_whisper/transcribe.py [hunk 1]
```diff
@@ -1449,6 +1602,32 @@ def key_func(language):
         # Language is not detected for any segment and none of prev conditions met
         return {'language_code': 'silence','language_confidence': 1.0} 
 
+default_batched_asr_options =  {
```
Comment (movchan74) [inline faster_whisper/transcribe.py:1605]: Wouldn't it be better to take it from the function's defaults like it's done in the official faster-whisper? https://github.com/SYSTRAN/faster-whisper/blob/1eb9a8004c509a4af2960955374520c35b7b793a/faster_whisper/transcribe.py#L184

File: faster_whisper/transcribe.py [hunk 2]
```diff
@@ -218,77 +198,61 @@ def get_language_and_tokenizer(self, audio, task=None, language=None):
...
 
-    def transcribe_stream(
-        self, audio: Union[str, np.ndarray], vad_segments, batch_size=None, num_workers=0, language=None, task=None, log_progress = False, combined_progress=False
-    ) -> TranscriptionResult:
+    def transcribe(
+        self, audio: Union[str, np.ndarray], vad_segments: List[dict], batch_size=16, num_workers=0, language=None, task=None, log_progress = False
+    ) ->Tuple[Iterable[BatchedSegment], dict]: 
+        """transcribe audio in chunks specified in vad_segments and return BatchedSegment generator with language info.
+
+        args:
```
Comment (movchan74) [inline faster_whisper/transcribe.py:212]: The style of the docstring is a bit different from the rest of the code. Return is not documented.

File: faster_whisper/transcribe.py [hunk 3]
```diff
@@ -218,77 +198,61 @@ def get_language_and_tokenizer(self, audio, task=None, language=None):
...
             f1 = int(seg['start'] * sampling_rate)
             f2 = int(seg['end'] * sampling_rate)
             yield {'inputs': audio[f1:f2]}
 
-    def transcribe_stream(
-        self, audio: Union[str, np.ndarray], vad_segments, batch_size=None, num_workers=0, language=None, task=None, log_progress = False, combined_progress=False
-    ) -> TranscriptionResult:
+    def transcribe(
+        self, audio: Union[str, np.ndarray], vad_segments: List[dict], batch_size=16, num_workers=0, language=None, task=None, log_progress = False
+    ) ->Tuple[Iterable[BatchedSegment], dict]:
```
Comment (movchan74) [inline faster_whisper/transcribe.py:209]: What's in the dict? Is there a TypedDict defined for it?

File: faster_whisper/transcribe.py [hunk 4]
```diff
@@ -218,77 +198,61 @@ def get_language_and_tokenizer(self, audio, task=None, language=None):
...
+            task: either "transcribe" or "translate".
+            log_progress: whether to show progress bar or not. 
+        """
+        sampling_rate = self.model.feature_extractor.sampling_rate
+
         if isinstance(audio, str):
             audio = decode_audio(audio)
 
+        #if no segment split is provided, consider a single vad segment from start to end. No benefit in this case.
+        if not vad_segments:
```
Comment (movchan74) [inline faster_whisper/transcribe.py:227]: If no VAD segments is an option then it should be documented and vad_segments  should be optional then.

**Review by movchan74 — COMMENTED** [review]
At least fix the documentation style.

File: faster_whisper/transcribe.py (hunk 1 above)
Comment (Jiltseb) [reply faster_whisper/transcribe.py:1605]: There will be a slightly lesser number of parameters for the batched version as well and then pass through the iterator to the forward function. Then it should support all latest additional parameters of the transcribe function but just do not allow static values for batched whisper (word timestamps as a parameter for example). Whats your take on it?

File: faster_whisper/transcribe.py (hunk 3 above)
Comment (Jiltseb) [reply faster_whisper/transcribe.py:209]: As you can see in the yield line, it is a dictionary containing language and its probability. There is no TypedDict defined only for it.

File: faster_whisper/transcribe.py (hunk 4 above)
Comment (Jiltseb) [reply faster_whisper/transcribe.py:227]: It was not an option before, rather a required parameter. For efficient working of the batched whisper, you need non empty vad_segments. If empty/None comes we should gracefully still run it. 

Or do you think adding the vad model inside and handling it inside is better?

**Review by evanderiel — APPROVED** [review]

File: faster_whisper/transcribe.py (hunk 1 above)
Comment (Jiltseb) [reply faster_whisper/transcribe.py:1789]: Updated the transcribe function and modified pipeline to allow run time parameters.

File: faster_whisper/transcribe.py (hunk 2 above)
Comment (Jiltseb) [reply faster_whisper/transcribe.py]: Updated

File: faster_whisper/transcribe.py (hunk 3 above)
Comment (Jiltseb) [reply faster_whisper/transcribe.py]: Uses TranscriptionInfo object instead of the dict.

File: faster_whisper/transcribe.py (hunk 4 above)
Comment (Jiltseb) [reply faster_whisper/transcribe.py:369]: Added optional and docstring for vad_segments

**Review by evanderiel — APPROVED** [review]

**Pull request merged**